meson.build

```diff
@@ -0,0 +1,439 @@
+  'Source/C++/Core/Ap4Processor.h',
+  'Source/C++/Core/Ap4Protection.h',
+  'Source/C++/Core/Ap4PsshAtom.h',
+  'Source/C++/Core/Ap4Results.h',
+  'Source/C++/Core/Ap4RtpAtom.h',
+  'Source/C++/Core/Ap4RtpHint.h',
+  'Source/C++/Core/Ap4SaioAtom.h',
+  'Source/C++/Core/Ap4SaizAtom.h',
+  'Source/C++/Core/Ap4Sample.h',
+  'Source/C++/Core/Ap4SampleDescription.h',
+  'Source/C++/Core/Ap4SampleEntry.h',
+  'Source/C++/Core/Ap4SampleSource.h',
+  'Source/C++/Core/Ap4SampleTable.h',
+  'Source/C++/Core/Ap4SbgpAtom.h',
+  'Source/C++/Core/Ap4SchmAtom.h',
+  'Source/C++/Core/Ap4SdpAtom.h',
+  'Source/C++/Core/Ap4SegmentBuilder.h',
+  'Source/C++/Core/Ap4SencAtom.h',
+  'Source/C++/Core/Ap4SgpdAtom.h',
+  'Source/C++/Core/Ap4SidxAtom.h',
+  'Source/C++/Core/Ap4SLConfigDescriptor.h',
+  'Source/C++/Core/Ap4SmhdAtom.h',
+  'Source/C++/Core/Ap4StcoAtom.h',
+  'Source/C++/Core/Ap4SthdAtom.h',
+  'Source/C++/Core/Ap4String.h',
+  'Source/C++/Core/Ap4StscAtom.h',
+  'Source/C++/Core/Ap4StsdAtom.h',
+  'Source/C++/Core/Ap4StssAtom.h',
+  'Source/C++/Core/Ap4StszAtom.h',
+  'Source/C++/Core/Ap4SttsAtom.h',
+  'Source/C++/Core/Ap4Stz2Atom.h',
+  'Source/C++/Core/Ap4SyntheticSampleTable.h',
+  'Source/C++/Core/Ap4TencAtom.h',
+  'Source/C++/Core/Ap4TfdtAtom.h',
+  'Source/C++/Core/Ap4TfhdAtom.h',
+  'Source/C++/Core/Ap4TfraAtom.h',
+  'Source/C++/Core/Ap4TimsAtom.h',
+  'Source/C++/Core/Ap4TkhdAtom.h',
+  'Source/C++/Core/Ap4Track.h',
+  'Source/C++/Core/Ap4TrakAtom.h',
+  'Source/C++/Core/Ap4TrefTypeAtom.h',
+  'Source/C++/Core/Ap4TrexAtom.h',
+  'Source/C++/Core/Ap4TrunAtom.h',
+  'Source/C++/Core/Ap4Types.h',
+  'Source/C++/Core/Ap4UrlAtom.h',
+  'Source/C++/Core/Ap4Utils.h',
+  'Source/C++/Core/Ap4UuidAtom.h',
+  'Source/C++/Core/Ap4Version.h',
+  'Source/C++/Core/Ap4VmhdAtom.h',
+  'Source/C++/Core/Ap4VpccAtom.h',
+  'Source/C++/Crypto/Ap4AesBlockCipher.h',
+  'Source/C++/Crypto/Ap4Hmac.h',
+  'Source/C++/Crypto/Ap4KeyWrap.h',
+  'Source/C++/Crypto/Ap4StreamCipher.h',
+  'Source/C++/MetaData/Ap4MetaData.h'
+]
+
+ap4_include = [
+  include_directories('Source/C++/Codecs'),
+  include_directories('Source/C++/Core'),
+  include_directories('Source/C++/Crypto'),
+  include_directories('Source/C++/MetaData')
+]
+
+if get_option('default_library') == 'both'
+  error('default library cannot be set to both, it can be either static or shared')
+endif
+
+if meson.get_compiler('cpp').get_id() == 'msvc' and get_option('default_library') == 'shared'
+  warning('building shared library with msvc doesn\'t export any symbols')
+endif
+
+ap4_lib = library(
+  'ap4',
+  ap4_sources,
+  install: true,
+  include_directories: ap4_include,
+)
+
+project_dep = declare_dependency(
+  include_directories: ap4_include,
+  link_with: ap4_lib
+)
+set_variable(meson.project_name() + '_dep', project_dep)
+
+install_headers(
+  ap4_headers + [
+    'Source/C++/Adapters/Ap4AtomixAdapters.h',
+    'Source/C++/Adapters/Ap4NeptuneAdapters.h',
+    'Source/C++/CApi/Bento4C.h'
+  ]
+)
```

Comment:
The cmake files install headers into the equivalent of adding `subdir: 'bento4'` here.

Reply:
Headers file should be kept under a subdirectory  bento4. I was trying to follow dir structure used [here](https://www.bento4.com/downloads). It is better to keep them in a separate directory. FIXED